src/main/java/com/exasol/dbbuilder/objectwriter/ImmediateDatabaseObjectWriter.java

```diff
@@ -170,4 +208,84 @@ public void write(final VirtualSchema virtualSchema) {
         }
         writeToObject(virtualSchema, builder.toString());
     }
+
+    @Override
+    public int execute(final Script script, final Object... parameterValues) {
+        try (final Statement statement = this.connection.createStatement()) {
+            statement.execute(getScriptExecutionSql(script, parameterValues));
+            return statement.getUpdateCount();
+        } catch (final SQLException exception) {
+            throw new DatabaseObjectException(script, "Failed to execute script " + script.getFullyQualifiedName(),
+                    exception);
```

Comment:
I would also add an exception cause massage to the string

Reply:
It's often not displayed at all, just a trace appears

Reply:
This is run outside of the VS, in test code only so the trace should contain the cause's message. At least it did in my tests.